src/lib/nft/Base64Storage.js

```diff
@@ -0,0 +1,70 @@
+import { memoize } from 'lodash'
+
+import Config from '../../config/config'
+import { fallback } from '../utils/async'
+import { Blob, NFTStorage } from './client'
+
+const { nftStorageKey, nftPeers } = Config
+
+class Base64Storage {
+  constructor(apiKey, peers) {
+    this.client = new NFTStorage({ token: apiKey })
+    this.peers = peers
+  }
+
+  // eslint-disable-next-line require-await
+  async store(base64) {
+    const { client } = this
+    const blob = new Blob([base64])
+
+    return client.storeBlob(blob)
+  }
+
+  async load(cid, skipCache = false) {
+    const clear = () => this._clearCache(cid)
+
+    if (true === skipCache) {
+      clear()
+    }
+
+    try {
+      return await this._load(cid)
+    } catch (exception) {
+      clear()
+      throw exception
+    }
+  }
+
+  // eslint-disable-next-line require-await
+  async delete(cid) {
+    const { client } = this
+
+    return client.delete(cid)
+  }
+
+  // eslint-disable-next-line require-await
+  _load = memoize(async cid => {
+    const { peers } = this
+
+    return fallback(
+      peers.map(async peer => {
```

Comment:
why not use Promise.race for peers?

Reply:
that would be wasteful.
just if a peer is down we should try another

Reply:
it's a fallback algorithm . we won't to query all at time
if one request failed - we're moving to another one peer until got success/traverse all